src/groups/mqb/mqbs/mqbs_filestore.cpp

```diff
@@ -4844,12 +4844,23 @@ void FileStore::replicateRecord(bmqp::StorageMessageType::Enum type,
         journalRecordBufferSp,
         FileStoreProtocol::k_JOURNAL_RECORD_SIZE);
 
-    bmqt::EventBuilderResult::Enum buildRc = d_storageEventBuilder.packMessage(
-        type,
-        d_config.partitionId(),
-        0,  // flags
-        journalOffsetWords,
-        journalRecordBlobBuffer);
+    bmqt::EventBuilderResult::Enum buildRc;
+    bool                           doRetry = false;
+    do {
+        buildRc = d_storageEventBuilder.packMessage(type,
+                                                    d_config.partitionId(),
+                                                    0,  // flags
+                                                    journalOffsetWords,
+                                                    journalRecordBlobBuffer);
+        if (buildRc == bmqt::EventBuilderResult::e_EVENT_TOO_BIG && !doRetry) {
+            flushIfNeeded(true);
+
+            doRetry = true;
+        }
+        else {
+            doRetry = false;
+        }
+    } while (doRetry);
```

Comment:
Just an idea:
- get rid of a loop
- get rid of tmp `bool doRetry`
- add performance hint
- as a bonus, in a debugger you will see that you are currently on the first try or on the second

```suggestion
    buildRc = d_storageEventBuilder.packMessage(type,
                                                d_config.partitionId(),
                                                0,  // flags
                                                journalOffsetWords,
                                                journalRecordBlobBuffer);
    if (BSLS_PERFORMANCEHINT_PREDICT_UNLIKELY(buildRc == bmqt::EventBuilderResult::e_EVENT_TOO_BIG)) {
        BSLS_PERFORMANCEHINT_UNLIKELY_HINT;
        flushIfNeeded(true);
        buildRc = d_storageEventBuilder.packMessage(type,
                                                    d_config.partitionId(),
                                                    0,  // flags
                                                    journalOffsetWords,
                                                    journalRecordBlobBuffer);
    }
```

Reply:
I would prefer a version calling `packMessage ` once, loop is ok, `BSLS_PERFORMANCEHINT_PREDICT_UNLIKELY` is a good idea